soccer.rst

```diff
@@ -0,0 +1,146 @@
+.. _SoccerIPM:
+
+Soccer IPM
+##########
+
+Overview
+========
+
+Soccer IPM is a ROS package that provides a node to project 2D image space detections of soccer related objects (e.g., balls, goal posts, robots, etc.) to 3D world space. 
+The node uses the camera calibration parameters as well as the forward kinematics of the robot to perform the projection.
+It is assumed that all objects are on the ground plane.
+The input detections are expected to be in the form of `soccer_vision_2d_msgs <https://github.com/ros-sports/soccer_interfaces/tree/rolling/soccer_vision_2d_msgs/msg>`_ messages.
+The projected detections are published in the form of `soccer_vision_3d_msgs <https://github.com/ros-sports/soccer_interfaces/tree/rolling/soccer_vision_3d_msgs/msg>`_ messages as defined in the `soccer_interfaces <https://github.com/ros-sports/soccer_interfaces>`_.
+
+You need to provide it also with some parameters such as the ball diameter, output frame, etc..
+
+
+Subscriptions
+-------------
+
+- ``camera_info``: Subscribes to camera info.
+- ``balls_in_image``: Subscribes to ball detections in image space. 
+- ``goal_posts_in_image``: Subscribes to goal post detections in image space.
+- ``robots_in_image``: Subscribes to robot detections in image space.
+- ``obstacles_in_image``: Subscribes to obstacle detections in image space.
+- ``field_boundary_in_image``: Subscribes to the detected field boundary in image space. 
+- ``markings_in_image``: Subscribes to line marking detections in image space.
+- ``tf``: Subscribes to tf updates.
+- ``tf_static``: Subscribes to tf static information.
+
+Publications
+------------
+
+- ``balls_relative``: Publishes the detected balls in the given output frame.
+- ``goal_posts_relative``: Publishes the detected goalposts in the given output frame.
+- ``robots_relative``: Publishes the detected robots in the given output frame.
+- ``obstacles_relative``: Publishes the detected obstacles in the given output frame.
+- ``field_boundary_relative``: Publishes the detected field boundary in the given output frame.
+- ``markings_relative``: Publishes the detected line markings in the given output frame.
+
+
+How do I use it?
+================
+
+First of all, you need to download the repo as well as its dependencies and build the package in your workspace.
+You can do this by running the following commands:
+
+.. code-block:: bash
+
+   git clone https://github.com/ros-sports/soccer_ipm.git src/soccer_ipm
+   rosdep install --from-paths src --ignore-src -r -y
```

Comment:
rosdep might need a little bit more setup. We could probably just reference a guide for this

Reply:
We don't do this in the other guides